packages/gatsby-plugin-gatsby-cloud/src/create-redirects.js

```diff
@@ -4,15 +4,12 @@ import { REDIRECTS_FILENAME } from "./constants"
 export default async function writeRedirectsFile(
   pluginData,
   redirects,
-  rewrites,
-  pathPrefix
+  rewrites
 ) {
   const { publicFolder } = pluginData
 
-  if (!redirects.length && !rewrites.length) return null
-
   // gatsby adds path-prefix to redirects so we need to remove them again
-  if (redirects && pluginData.pathPrefix) {
+  if (redirects && redirects.length && pluginData.pathPrefix) {
```

Comment:
It's always an array so the redirects.map will be a no-op anyway.
```suggestion
  if (redirects && pluginData.pathPrefix) {
```